### PR TITLE
Updated XML-doc comments around CodeDomComponentSerializationService

### PR DESCRIPTION
**Pull request by Tanya-Solyanik**
Related to https://github.com/dotnet/winforms/issues/10425

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/11166)

### CODE REVIEW

File: src/System.Windows.Forms.Design/src/System/ComponentModel/Design/Serialization/CodeDomComponentSerializationService.cs [hunk 1]
```diff
@@ -1,16 +1,18 @@
...
 
 /// <summary>
-///  This class serializes a set of components or serializable objects into a serialization store. The store can then
-///  be deserialized at a later date. CodeDomComponentSerializationService differs from other serialization schemes in
+///  On .NET Framework, this class serializes a set of components or serializable objects into a serialization store. The store can then
+///  be deserialized at a later time. The <see cref="CodeDomComponentSerializationService"/> differs from other serialization schemes in
 ///  that the serialization format is opaque, and it allows for partial serialization of objects. For example, you can
 ///  choose to serialize only selected properties for an object.
+///  On .NET, <see cref="CodeDomComponentSerializationService"/> can create a <see cref="SerializationStore"/> in memory,
+///  but that store can't be saved to a stream , not can a new store be loaded from a stream, as the
```
Comment (lonitra) [inline src/System.Windows.Forms.Design/src/System/ComponentModel/Design/Serialization/CodeDomComponentSerializationService.cs:14]: Did you mean the following instead?
```suggestion
///  but that store can't be saved to a stream , nor can a new store be loaded from a stream, as the
```

File: src/System.Windows.Forms.Design/src/System/ComponentModel/Design/Serialization/CodeDomComponentSerializationService.cs [hunk 2]
```diff
@@ -24,31 +26,32 @@ public CodeDomComponentSerializationService() : this(null)
     }
 
     /// <summary>
-    ///  Creates a new CodeDomComponentSerializationService object using the given  service provider to resolve services.
+    ///  Creates a new <see cref="CodeDomComponentSerializationService"/> object using the given  service provider to resolve services.
```
Comment (lonitra) [inline src/System.Windows.Forms.Design/src/System/ComponentModel/Design/Serialization/CodeDomComponentSerializationService.cs:29]:
```suggestion
    ///  Creates a new <see cref="CodeDomComponentSerializationService"/> object using the given service provider to resolve services.
```

File: test.cmd [hunk 3]
```diff
@@ -0,0 +1,3 @@
+@echo off
```
Comment (lonitra) [inline test.cmd:1]: Why was this added?

Comment (Tanya-Solyanik) [reply test.cmd:1]: This is a convenience script to run tests from the command line by typing a little less, and more importantly it matches the designer repo workflow.

File: src/System.Windows.Forms.Design/src/System/ComponentModel/Design/Serialization/CodeDomComponentSerializationService.CodeDomSerializationStore.cs [hunk 4]
```diff
@@ -294,22 +297,15 @@ private static string GetObjectName(object value)
...
-
-        /// <summary>
-        ///  The Save method is not supported.
+        ///  The <see cref="Save"/> method is not supported on .NET.
         /// </summary>
         public override void Save(Stream stream) => throw new PlatformNotSupportedException();
 
         /// <summary>
-        ///  Implements the save part of ISerializable. Used in unit tests only.
+        ///  On .NET Framework, this method implements the save part of <see cref="ISerializable"/>. On .NET, this interface
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms.Design/src/System/ComponentModel/Design/Serialization/CodeDomComponentSerializationService.CodeDomSerializationStore.cs:305]:
```suggestion
        ///  On .NET Framework, this method implements the save part of <see cref="ISerializable"/>.  On .NET, this interface
```

File: src/System.Windows.Forms.Design/src/System/ComponentModel/Design/Serialization/CodeDomComponentSerializationService.CodeDomSerializationStore.cs [hunk 5]
```diff
@@ -146,7 +149,7 @@ public override void Close()
                 _errors = manager.Errors;
             }
 
-            // also serialize out resources if we have any  we force this in order for undo to work correctly
+            // Also serialize out resources if we have any  we force this in order for undo to work correctly.
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms.Design/src/System/ComponentModel/Design/Serialization/CodeDomComponentSerializationService.CodeDomSerializationStore.cs:152]:
```suggestion
            // Also serialize out resources if we have any we force this in order for undo to work correctly.
```

File: src/System.Windows.Forms.Design/src/System/ComponentModel/Design/Serialization/CodeDomComponentSerializationService.CodeDomSerializationStore.cs [hunk 6]
```diff
@@ -114,7 +116,8 @@ internal void AddObject(object value, bool absolute)
         }
 
         /// <summary>
-        ///  The Close method closes this store and prevents any objects  from being serialized into it.  Once closed, the serialization store may be saved.
+        ///  The <see cref="Close"/> method closes this store and prevents any objects from being serialized into it.  Once closed,
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms.Design/src/System/ComponentModel/Design/Serialization/CodeDomComponentSerializationService.CodeDomSerializationStore.cs:119]:
```suggestion
        ///  The <see cref="Close"/> method closes this store and prevents any objects from being serialized into it. 
```

File: src/System.Windows.Forms.Design/src/System/ComponentModel/Design/Serialization/CodeDomComponentSerializationService.CodeDomSerializationStore.cs [hunk 7]
```diff
@@ -114,7 +116,8 @@ internal void AddObject(object value, bool absolute)
...
+        ///  the serialization store may be saved.
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms.Design/src/System/ComponentModel/Design/Serialization/CodeDomComponentSerializationService.CodeDomSerializationStore.cs:120]:
```suggestion
```

**Review by lonitra — APPROVED** [review]
